libpimeval/src/pimPerfEnergyModels.cpp

```diff
@@ -10,8 +10,40 @@
 #include "pimPerfEnergyTables.h"
 #include <cstdio>
 
-//! @brief  pimParamsPerf ctor
-pimParamsPerf::pimParamsPerf(pimParamsDram* paramsDram)
+
+//! @brief  A factory function to get perf energy model for sim target
+pimPerfEnergyBase*
+pimPerfEnergyFactory::getPerfEnergyModel(PimDeviceEnum simTarget, pimParamsDram* paramsDram)
```

Comment:
```suggestion
pimPerfEnergyFactory::getPerfEnergyModel(PimDeviceEnum simTarget, const pimParamsDram& paramsDram)
```
This is to avoid accidental changes to the parameters